### ISSUE TITLE
Provide defaults for `useUser()`'s generic type parameters

### ISSUE THREAD

**Issue by elle-j** [realm/realm-js#6764]
If you want to provide e.g. a type for the Atlas App Services functions defined (generic type param 1), currently you'd also need to provide types for the remaining generic type params:

```ts
const user = useUser<MyAppServicesFunctions, Record<string, unknown>, Realm.DefaultUserProfileData>();
```

We could provide defaults to reduce the verbosity:

```ts
const user = useUser<MyAppServicesFunctions>();
```

**Comment by kraenhansen** [comment]
I agree - this feels like an oversight 👍

**Comment by vinibgoulart** [comment]
Can we close this? Already fixed in this PR: #6814
